app/controllers/api/v1/users_controller.rb

```diff
@@ -1,14 +1,36 @@
 # frozen_string_literal: true
 
+require 'securerandom'
 module Api
   module V1
     class UsersController < AuthenticatedController
-      before_action :require_authentication, except: [:recover_password, :create]
-      before_action :set_user, only: [:recover_password, :show, :update, :destroy]
+      before_action :require_authentication, except: [:recover_password, :recovery_token, :create]
+      before_action :set_user, only: [:show, :update, :destroy]
       before_action :require_authorisation, only: [:update, :destroy]
+      before_action :check_token, only: [:recover_password]
 
       def recover_password
-        PasswordRecoveryMailer.with(user: @user).recover_password.deliver_later
+        @user.password = SecureRandom.base64(10)
+        @user.token = nil
+
+        if @user.save
+          PasswordRecoveryMailer.with(user: @user).recover_password.deliver_later
```

Comment:
You should store the generated password in a variable and send it here separately - in a normal system you wouldn't be able to access it from `user.password` 😅